File: app/src/main/java/com/example/pearvideoclient/follow/FollowUserListAdapter.java
package com.example.pearvideoclient.follow;

import android.support.annotation.Nullable;
import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.pearvideoclient.R;
import com.example.pearvideoclient.entity.MyFollowContBean;
import com.example.pearvideoclient.utils.GlideUtils;

import java.util.List;
import java.util.Objects;

import static com.example.pearvideoclient.follow.FollowPresenter.MORE_USER;

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-10 21:00
 * @ClassName: FollowUserListAdapter
 */
public class FollowUserListAdapter extends BaseQuickAdapter<MyFollowContBean.FollowUserListBean, BaseViewHolder> {
    FollowUserListAdapter(int layoutResId, @Nullable List<MyFollowContBean.FollowUserListBean> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, MyFollowContBean.FollowUserListBean item) {
        helper.setText(R.id.tv_user_name, item.getNickname());
        ImageView ivUserImage = helper.getView(R.id.iv_user_img);
        if (MORE_USER.equals(item.getUserId())) {
            GlideUtils.load(Objects.requireNonNull(mContext.getDrawable(R.drawable.ic_arrow_right)), ivUserImage);
        } else {
            GlideUtils.loadCircleImage(item.getPic(), ivUserImage);
        }

        helper.addOnClickListener(R.id.rl_parent);
    }
}

File: app/src/main/java/com/example/pearvideoclient/follow/FollowUserListAdapter.kt
package com.example.pearvideoclient.follow

import android.graphics.drawable.Drawable
import android.widget.ImageView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.pearvideoclient.R
import com.example.pearvideoclient.entity.MyFollowContBean
import com.example.pearvideoclient.follow.FollowPresenter.Companion.MORE_USER
import com.example.pearvideoclient.utils.GlideUtils
import java.util.*


/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-10 21:00
 * @ClassName: FollowUserListAdapter
 */
class FollowUserListAdapter internal constructor(layoutResId: Int, data: List<MyFollowContBean.FollowUserListBean>?)
    : BaseQuickAdapter<MyFollowContBean.FollowUserListBean, BaseViewHolder>(layoutResId, data) {

    override fun convert(helper: BaseViewHolder, item: MyFollowContBean.FollowUserListBean) {
        helper.setText(R.id.tv_user_name, item.nickname)
        val ivUserImage = helper.getView<ImageView>(R.id.iv_user_img)
        if (MORE_USER == item.userId) {
            GlideUtils.load(Objects.requireNonNull<Drawable>(mContext.getDrawable(R.drawable.ic_arrow_right)), ivUserImage)
        } else {
            GlideUtils.loadCircleImage(item.pic, ivUserImage)
        }

        helper.addOnClickListener(R.id.rl_parent)
    }
}
